Fixes a fuck ton more harddels

LemonInTheDark opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
 Redoes how geese handle eating shit, it was fucking stupid and caused harddels, and while this method is technically slower in the best case, it's a fucking goose
Fixes action related harddels, I hate how they work but at least this way they won't hold refs.
Fixes the hierophont causing its beacon to harddel
Removes the M variable from megafauna actions, it was used like a typed owner and caused harddels, so I burned it
Fixes target and targets_from harddels, replaces all setters of target with LoseTarget and GiveTarget, which should help maintain behavior. I'm not sure if this breaks anything, but if it does we should fix the assumptions that code makes instead of reverting this change
Fixes more area_senstive_contents related harddels, we need to allow the mob to move before clearing out its list.
Fixes marked object harddels (I'm coming for you admin team)
Fixes a language based human harddel
Fixes managed overlay related harddels (This was just emissive blockers, but I think this is a good safety net to have. If we clear the overlay list we should clear this one as well)
Fixes bot core harddels, I hate the fact that this exists but it has no reason to know who its owner is
 Adds a walk(src, 0) to simple_animal destroy, it's the best bang for the buck in terms of stopping spurious harddels. Walk related harddels aren't that expensive in the first place, since byond does the same thing I'm doing here, but this makes finding mob harddels easier, so let's go with it
I fixed another source of part harddels, I hate fullupgrade so much
Fixes all the sound loop harddels 

## Why It's Good For The Game

I want more processing time god damn it

## Changelog
:cl:
code: Td should spike slightly less, assuming harddels actually go down
code: You feel a great disturbance in the force, mob code has changed. Please report any odd targeting behavior on the issue tracker
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

LemonInTheDark committed:
temp

LemonInTheDark committed:
Redoes how geese handle eating shit, it was fucking stupid and caused harddels, and while this method is technically slower in the best case, it's a fucking goose

LemonInTheDark committed:
Fixes action realted harddels, I hate this so much

LemonInTheDark committed:
Fixes the hierophont causing its beacon to harddel

LemonInTheDark committed:
Removes the M variable from megafauna actions, it was used like a typed owner, and for that reason died in a fire

LemonInTheDark committed:
Fixes target and targets_from harddels, replaces all setters of target with LoseTarget and GiveTarget, which should help maintain behavior

LemonInTheDark committed:
Fixes more area_senstive_contents related harddels

LemonInTheDark committed:
Fixes marked object harddels (I'm coming for you admin team

LemonInTheDark committed:
Fixes a language based human harddel

LemonInTheDark committed:
Fixes managed overlay related harddels

LemonInTheDark committed:
Fixes bot core harddels, I hate the fact that this exists but it has no reason to know who its owner is

LemonInTheDark committed:
Adds a walk(src, 0) to simple_animal destroy, it's the best bang for buck in terms of stopping spurious harddels. Walk related harddels aren't that expensive in the first place, since byond does the same thing I'm doing here, but this makes finding mob harddels easier, so let's go with it

LemonInTheDark committed:
Removes my debugging env, oh I also fixed another source of part harddels, I hate fullupgrade

interestingusernam3 commented:
‏

Mothblocks commented on code/datums/action.dm:
```diff
@@ -72,13 +73,18 @@
 	else
 		Remove(owner)
 
+/datum/action/proc/owner_deleted(datum/source)
```
SIGNAL_HANDLER

LemonInTheDark replied:
gottchu

Mothblocks commented on code/modules/mob/living/simple_animal/bot/bot.dm:
```diff
@@ -897,11 +897,10 @@ Pass a positive integer as an argument to override a bot's default speed.
 /obj/machinery/bot_core/Initialize()
 	. = ..()
-	owner = loc
+	var/mob/living/simple_animal/bot/owner = loc
 	if(!istype(owner))
```
```suggestion
	if(!isbot(loc))
```

Mothblocks commented on code/modules/mob/living/simple_animal/hostile/megafauna/megafauna.dm:
```diff
@@ -179,16 +179,15 @@
-		M = L
-		return ..()
-	return FALSE
+	if(!istype(L, /mob/living/simple_animal/hostile/megafauna))
```
```suggestion
	if(!ismegafauna(L))
```

LemonInTheDark committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into moreharddelstuff

LemonInTheDark committed:
Some comments, code cleanup, and moth's review

LemonInTheDark committed:
Fixes all the sound loop harddels

LemonInTheDark committed:
Cleans up the plumbing component's del react a bit, the source of the signal was being read as reveiver, which
of course caused harddels
Makes a qdel() -> QDEL_NULL, seems like a good idea

LemonInTheDark committed:
Solves a doozy of an issue.
A duct was placed underneath a showerhead at roundstart, so when the shower initialized it broke the duct,
qdeling it.
the duct was still scheculed for lateInit tho, and because late init creates area_sensitive_contents, it would
ref itself, and harddel. This solves the problem by preventing qdeleted objects from late initing
